src/onboarding/restore-backup/onboarding-restore-backup-restore.ts

```diff
@@ -0,0 +1,179 @@
+import { css, html, LitElement, nothing, type CSSResultGroup } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../components/ha-card";
+import "../../components/ha-alert";
+import "../../components/buttons/ha-progress-button";
+import "../../components/ha-password-field";
+import { haStyle } from "../../resources/styles";
+import type { LocalizeFunc } from "../../common/translations/localize";
+import {
+  CORE_LOCAL_AGENT,
+  HASSIO_LOCAL_AGENT,
+  type BackupContentExtended,
+  type BackupData,
+} from "../../data/backup";
+import { restoreOnboardingBackup } from "../../data/backup_onboarding";
+import type { HaProgressButton } from "../../components/buttons/ha-progress-button";
+import { fireEvent } from "../../common/dom/fire_event";
+
+declare global {
+  interface HASSDomEvents {
+    "restore-started";
+  }
+}
+@customElement("onboarding-restore-backup-restore")
+class OnboardingRestoreBackupRestore extends LitElement {
+  @property({ attribute: false }) public localize!: LocalizeFunc;
+
+  @property({ type: Object }) public backup!: BackupContentExtended;
+
+  @property({ type: Object, attribute: false })
+  public selectedData!: BackupData;
+
+  @property({ type: Boolean }) public supervisor = false;
+
+  @state() private _encryptionKey = "";
+
+  @state() private _encryptionKeyWrong = false;
+
+  @state() private _error?: string;
+
+  @state() private _loading = false;
+
+  render() {
+    const agentId = this.supervisor ? HASSIO_LOCAL_AGENT : CORE_LOCAL_AGENT;
+    const backupProtected = this.backup.agents[agentId].protected;
+
+    return html`
+      ${this.backup.homeassistant_included &&
+      !this.supervisor &&
+      (this.backup.addons.length > 0 || this.backup.folders.length > 0)
+        ? html`<div class="supervisor-warning">
+            <ha-alert alert-type="warning">
+              ${this.localize(
+                "ui.panel.page-onboarding.restore.details.backup_unsupported"
```

Comment:
This translation key makes it look like it will not work, but it will work, just not completly.